test/test_nn.py

```diff
@@ -1600,6 +1600,21 @@ def test_Sequential_append(self):
         self.assertEqual(n2, nn.Sequential(l1, l2, l3, l4))
         self.assertEqual(nn.Sequential(l1).append(l2).append(l4), nn.Sequential(l1, l2, l4))
 
+    def test_Sequential_pop(self):
+        l1 = nn.Linear(1, 2)
+        l2 = nn.Linear(2, 3)
+        l3 = nn.Linear(3, 4)
+        l4 = nn.Linear(4, 5)
+        n1 = nn.Sequential(l1, l2, l3, l4)
+        self.assertEqual(l4, n1.pop(3))
+        n2 = nn.Sequential(l1, l2, l3)
+        self.assertEqual(n1, n2)
+        # check order of the index
+        k = 0
+        for key in range(len(n1)):
+            self.assertEqual(key, k)
```

Comment:
I'm not sure this test is doing what you want as it compares the result of `range` with a manually incremented number.
I think doing something like this would be more reliable (to check both that the indices are good and the order is the same as the iterator).
```python
for k, mod in zip(range(len(n1)), n1):
  self.assertIs(n1[k], mod)
```

Reply:
Thanks, updated the test! This looks much better.